hyppo/independence/friedmanRafsky.py

```diff
@@ -0,0 +1,297 @@
+import random
+from numba import jit
+
+import numpy as np
+
+from .base import IndependenceTest, IndependenceTestOutput
+
+
+class FriedmanRafsky(IndependenceTest):
+
+    r"""
+    Friedman-Rafksy (FR) test statistic and p-value.
+    This is a multivariate extension of the Wald-Wolfowitz
+    runs test for randomness. The normal concept of a 'run'
+    is replaced by a minimum spanning tree calculated between
+    the points in respective data sets with edge weights defined
+    as the Euclidean distance between two such points. After MST
+    has been determined, all edges such that both corresponding
+    nodes do not belong to the same class are severed and the
+    number of independent resulting trees is counted. This test is
+    consistent against similar tests
+    :footcite:p:`GSARMultivariateAdaptationofWaldWolfowitzTest`
+    """
+
+    def __init__(self, **kwargs):
+
+        IndependenceTest.__init__(self, **kwargs)
+
+    class Graph:

```

Comment:
why is this a class in a class? This should be able to exist as a standalone class at the bottom of the file right?